test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.Cli.Sln.Internal;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using System;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.Add.Proj.Tests
+{
+    public class GivenDotnetAddProj : TestBase
+    {
+        [Theory]
+        [InlineData("--help")]
+        [InlineData("-h")]
+        public void WhenHelpOptionIsPassedItPrintsUsage(string helpArg)
+        {
+            var cmd = new DotnetCommand()
+                .ExecuteWithCapturedOutput($"add project {helpArg}");
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Theory]
+        [InlineData("idontexist.sln")]
+        [InlineData("ihave?inv@lid/char\\acters")]
+        public void WhenNonExistingSolutionIsPassedItPrintsErrorAndUsage(string solutionName)
+        {
+            var cmd = new DotnetCommand()
+                .ExecuteWithCapturedOutput($"add {solutionName} project p.csproj");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("Could not find");
+            cmd.StdOut.Should().Contain("Usage:");
+        }
+
+        [Fact]
+        public void WhenInvalidSolutionIsPassedItPrintsErrorAndUsage()
+        {
+            var projectDirectory = TestAssetsManager.CreateTestInstance("InvalidSolution")
+                                                    .WithLockFiles()
+                                                    .Path;
+
+            var projectToAdd = Path.Combine("Lib", "Lib.csproj");
+            var cmd = new DotnetCommand()
+                .WithWorkingDirectory(projectDirectory)
+                .ExecuteWithCapturedOutput($"add InvalidSolution.sln project {projectToAdd}");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("Invalid solution ");
+            cmd.StdOut.Should().Contain("Usage:");
+        }
+
+        [Fact]
+        public void WhenInvalidSolutionIsFoundItPrintsErrorAndUsage()
+        {
+            var projectDirectory = TestAssetsManager.CreateTestInstance("InvalidSolution")
```

Comment:
Consider TestAssets.cs instead of TestAssetsManager.cs. I'm putting together a PR that moves us over in bulk, but it may be helpful to you here to use the new impl.